server/src/modules-resolver.ts

```diff
@@ -5,20 +5,10 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import * as fs from 'fs';
-import * as paths from 'path';
-
 export function findPathToModule(dir: string, moduleName: string): string|undefined {
-    const stat = fs.statSync(dir)
-    if (stat.isDirectory()) {
-        const candidate = paths.resolve(dir, 'node_modules', moduleName)
-        if (fs.existsSync(candidate)) {
-            return candidate;
-        }
-    }
-    const parent = paths.resolve(dir, '..')
-    if (parent !== dir) {
-        return findPathToModule(parent, moduleName)
+    try {
+        return require.resolve(moduleName, { paths: [dir] });
```

Comment:
```suggestion
        return fs.dirname(require.resolve(`${moduleName}/package.json`, { paths: [dir] }));
```
Seems like it actually wants the folder and not the main file

Reply:
Actually no. The naming of the argument `moduleName` is confusing because it's not really a module name. The two call sites that use it pass something like:
 - `.bin/tsserver`
 - `typescript/lib/tsserver.js`
And the returned value is expected to be a full path to those scripts.